modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc

```diff
@@ -0,0 +1,73 @@
+
+It is possible to create read privileges that are based on properties of nodes.
+Each property-based privilege can only be restricted by a single property.
+To specify the property/value conditions of the read privilege the `pattern` syntax described below is used,
+for more information about read privilege syntax see xref:authentication-authorization/privileges-reads.adoc[read privilege] page.
+
+Adding property-based access control can incur a significant performance overhead in certain scenarios,
+see xref:authentication-authorization/limitations.adoc#property-based-access-control-limitations[Limitations] for more detailed information.
+To reduce the performance impact, we recommend using Block Storage as this is better optimised for the kind of read required for the resolution of property-based privileges.
+
+Factors which will exacerbate the performance impact of having property rules are:
+
+1. The number of properties on the nodes concerned (more properties = greater performance impact)
+2. The number of property-based privileges (more property-based privileges = greater performance impact).
+3. The type of the privilege: `TRAVERSE` property-based privileges have a greater performance impact than `READ` property-based privileges.
+
+For performance-critical scenarios it is recommended to design  xref:authentication-authorization/privileges-reads.adoc[privileges based upon labels].
+
+Pattern syntax:
+[source, syntax, role="noheader"]
+----
+([var][:label["|" ...]] "{" property: value "}")
+| (var[:label["|" ...]]) WHERE [NOT] var.property { = value | <> value | IS NULL | IS NOT NULL }
+| (var[:label["|" ...]] WHERE [NOT] var.property { = value | <> value | IS NULL | IS NOT NULL } )
+----
+[NOTE]
+====
+For more details about the syntax descriptions, see xref:database-administration/syntax.adoc[].
+====
+This pattern syntax are used in the xref:authentication-authorization/privileges-reads.adoc[read privileges] as this:
+
+[source, syntax, role="noheader"]
+----
+GRANT ... ON GRAPH ... FOR pattern TO ...
+----
+
+
+For example, to grant role `regularUser` the ability to `READ` property `address` on nodes with labels `Email` or `Website` where the domain is `exampledomain.com`:
+[source, syntax, role="noheader"]
+----
+GRANT READ { address } ON GRAPH * FOR (n:Email|Website) WHERE n.domain = 'exampledomain.com' TO regularUsers
+----
+If preferred the same can be expressed using this syntax as well:
+[source, syntax, role="noheader"]
+----
+GRANT READ { address } ON GRAPH * FOR (:Email|Website {domain: 'exampledomain.com'}) TO regularUsers
+----
+
+
+Granting `TRAVERSE` privilege on nodes with label `Email` where property `classification` is `NULL` to role `regularUsers`:
+[source, syntax, role="noheader"]
+----
+GRANT TRAVERSE ON GRAPH * FOR (n:Email) WHERE n.classification IS NULL TO regularUsers
+----
+
+The role `regularUsers` can be denied to read and traverse nodes where the property `classification` is not `UNCLASSIFIED`:
+
```

Comment:
```suggestion
.Denying permission to `READ` and `TRAVERSE` nodes where the property `classification` is different from `UNCLASSIFIED` to role `regularUsers`:
```

Reply:
Should the dot in the start be there? looks odd?

Reply:
Yes, this will be the caption of the example. When we agree on everything, I'll build it locally and fix it if there are any formatting issues.